src/PresentationalComponents/TopicsAdminTable/TopicsAdminTable.js

```diff
@@ -1,8 +1,196 @@
-import React from 'react';
+import './_TopicsAdminTable.scss';
 
-const TopicsAdminTable = () => {
+import React, { useCallback, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { TableToolbar, PrimaryToolbar, Main, PageHeader } from '@redhat-cloud-services/frontend-components';
+import routerParams from '@redhat-cloud-services/frontend-components-utilities/files/RouterParams';
+import { sortable, Table, TableBody, TableHeader } from '@patternfly/react-table';
+import { CheckCircleIcon, BanIcon, EditAltIcon, StarIcon } from '@patternfly/react-icons';
+import { Button, Title } from '@patternfly/react-core';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import Immutable from 'seamless-immutable';
+
+import * as AppActions from '../../AppActions';
+import Loading from '../Loading/Loading';
+import Failed from '../Loading/Failed';
+import MessageState from '../MessageState/MessageState';
+import messages from '../../Messages';
+import AddEditTopic from '../Modals/AddEditTopic';
+
+const TopicsAdminTable = ({ topicsFetchStatus, fetchTopics, topics, intl }) => {
+
+    const [cols] =  useState([
+        { title: intl.formatMessage(messages.title), transforms: [sortable] },
+        { title: intl.formatMessage(messages.tag), transforms: [sortable] },
+        { title: intl.formatMessage(messages.topicSlug), transforms: [sortable] },
+        { title: intl.formatMessage(messages.status), transforms: [sortable] },
+        { title: intl.formatMessage(messages.featured), transforms: [sortable] },
+        '', ''
+    ]);
+    const [rows, setRows] = useState([]);
+    const [actions, setActions] = useState([]);
+    const [sortBy, setSortBy] = useState({});
+    const [addEditTopicOpen, setAddEditTopicOpen] = useState(false);
+    const [selectedTopic, setSelectedTopic] = useState({});
+    const [topicsArray, setTopicsArray] = useState([]);
+
+    const onSort = useCallback((_event, index, direction) => {
+        setSortBy({ index, direction });
+        setTopicsArray(
+            Immutable.from(
+                topicsArray.asMutable().sort((a, b) =>
+                    a[index] > b[index] ? 1 : -1
+                )
+            )
+        );
+    }, [setSortBy, setTopicsArray, topicsArray]);
+
+    const hideTopics = async (topic) => {
```

Comment:
this doesn't need to be async or the try catch block (also could kill the notification)